components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AuthorizationCodeGrantHandler.java

```diff
@@ -294,6 +296,14 @@ private void revokeExistingAccessTokens(String tokenId, AuthzCodeDO authzCodeDO)
             throw new IdentityOAuth2Exception("User id not found for user: "
                     + authzCodeDO.getAuthorizedUser().getLoggableUserId(), e);
         }
+        String accessToken = OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
+                .getAccessTokenByTokenId(tokenId);
+        // Fetching AccessTokenDO from DB before revoking the token
+        AccessTokenDO accessTokenDO = null;
+        if (StringUtils.isNotBlank(accessToken)) {
+            accessTokenDO = OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
```

Comment:
there is a util method in OAuth2util class to get it, it will get from cache, then check in DB can you use `getAccessTokenDOFromTokenIdentifier`

Reply:
Fixed with 33f8ddecc5895779fb2893cccb3f528d456fed43